### PR TITLE
RIA-3376: PACT contract updated as per signed off API spec

### PR DESCRIPTION
**Pull request by divisathyan**
RIA-3376: PACT contract updated as per signed off API spec

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3376

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: target/pacts/hmcts-homeoffice_api.json [hunk 1]
```diff
@@ -6,11 +6,451 @@
         "name": "hmcts"
     },
     "interactions": [
+        {
```
Comment (raghera) [inline target/pacts/hmcts-homeoffice_api.json:9]: You should not check in anything in the /target folder.  Please remove.

File: src/contractTest/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/HomeOfficeConsumerTest.java [hunk 2]
```diff
@@ -72,7 +163,89 @@ public void should_get_appeal_details_with_ho_reference(MockServer mockServer) t
...
+        assertThat(response).hasNoNullFieldsOrProperties();
+        assertThat(response.getJSONObject("messageHeader")).isNotNull();
+        assertThat(response.getJSONObject("messageHeader").getString("correlationId")).isNotBlank();
+        assertThat(response.getJSONObject("messageHeader").getString("eventDateTime")).isNotBlank();
+        assertThat(response.getJSONObject("messageHeader").getJSONObject("consumer")).isNotNull();
+        assertThat(response.getJSONObject("messageHeader").getJSONObject("consumer").getString("code"))
+            .isNotBlank();
+        assertThat(response.getJSONObject("messageHeader").getJSONObject("consumer").getString("description"))
+            .isNotBlank();
+        assertThat(response.getString("messageType")).isNotBlank();
```
Comment (raghera) [inline src/contractTest/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/HomeOfficeConsumerTest.java:187]: Could you check that the messageType is one of the valid Enum values here?  Would be a better check.  Same for any other Enums.

File: target/pacts/hmcts-homeoffice_api.json (hunk 1 above)
Comment (divisathyan) [reply target/pacts/hmcts-homeoffice_api.json:9]: Removed

File: src/contractTest/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/HomeOfficeConsumerTest.java (hunk 2 above)
Comment (divisathyan) [reply src/contractTest/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/HomeOfficeConsumerTest.java:187]: I have added StringMatcher in the contract and check in assert

**Pull request merged**